python/lsst/ts/atdome/dome_csc.py

```diff
@@ -566,34 +545,26 @@ async def connect(self):
                     port = self.mock_ctrl.port
                 else:
                     port = self.config.port
-                connect_coro = asyncio.open_connection(host=host, port=port)
-                self.reader, self.writer = await asyncio.wait_for(
-                    connect_coro, timeout=self.config.connection_timeout
+                self.client = tcpip.Client(host=host, port=port, log=self.log)
+                print("Constructed the client; waiting to connect")
+                await asyncio.wait_for(
+                    self.client.start_task, timeout=self.config.connection_timeout
                 )
+                print("Connected; wait for prompt")
                 # drop welcome message
                 await asyncio.wait_for(
-                    self.reader.readuntil(">".encode()),
-                    timeout=self.config.read_timeout,
+                    self.client.readuntil(b">"), timeout=self.config.read_timeout
                 )
+            print("Fully connected")
             self.log.debug("connected")
         except Exception as e:
-            err_msg = (
-                f"Could not open connection to host={host}, port={self.config.port}"
-            )
+            err_msg = f"Could not open connection to host={host}, port={self.config.port}: {e!r}"
```

Comment:
`port={self.config.port}` -> `port={port}`? In lines 545 and 547 you set that local variable.

Reply:
Good catch. Also I should set port and host at the same point in the code and stop using salobj.LOCAL_HOST.